src/backend/gl/src/command.rs

```diff
@@ -324,46 +325,42 @@ impl RawCommandBuffer {
         slice
     }
 
-    fn update_blend_targets(&mut self, blend_targets: &Vec<pso::ColorBlendDesc>) {
+    fn update_blend_targets(&mut self, blend_targets: &[pso::ColorBlendDesc]) {
         let max_blend_slots = blend_targets.len();
+        if max_blend_slots == 0 {
+            return;
+        }
 
-        if max_blend_slots > 0 {
-            match self.cache.blend_targets {
-                Some(ref mut cached) => {
-                    if cached.len() < max_blend_slots {
-                        cached.resize(max_blend_slots, None);
-                    }
-                }
-                None => {
-                    self.cache.blend_targets = Some(vec![None; max_blend_slots]);
-                }
-            };
+        if self.cache.blend_targets.len() < max_blend_slots {
+            self.cache.blend_targets.resize(max_blend_slots, None);
         }
 
-        for (slot, blend_target) in blend_targets.iter().enumerate() {
+        let all_targets_same = blend_targets[1..].iter().all(|target| target == &blend_targets[0]);
+
+        if all_targets_same {
             let mut update_blend = false;
-            if let Some(ref mut cached_targets) = self.cache.blend_targets {
-                if let Some(cached_target) = cached_targets.get(slot) {
-                    match cached_target {
-                        &Some(ref cache) => {
-                            if cache != blend_target {
-                                update_blend = true;
-                            }
-                        }
-                        &None => {
-                            update_blend = true;
-                        }
-                    }
+            for cached_target in &mut self.cache.blend_targets {
+                if cached_target.as_ref() != Some(&blend_targets[0]) {
+                    *cached_target = Some(blend_targets[0]);
+                    update_blend = true;
                 }
+            }
+            if update_blend {
+                self.push_cmd(Command::BindAllBlendSlots(blend_targets[0]));
+            }
+        } else {
+            for (slot, blend_target) in blend_targets.iter().enumerate() {
+                let cached_target = &mut self.cache.blend_targets[slot];
+                let update_blend = match cached_target {
+                    Some(cache) => cache != blend_target,
+                    None => true,
+                };
 
                 if update_blend {
-                    cached_targets[slot] = Some(*blend_target);
+                    *cached_target = Some(*blend_target);
+                    self.push_cmd(Command::BindBlendSlot(slot as _, *blend_target));
```

Comment:
could you consider using `push_cmd_internal` instead of `self.push_cmd` to avoid `self` borrowing conflict with `cache.blend_targets`?

Reply:
oh, good point